Week1/homework/app.js

```diff
@@ -1,11 +1,182 @@
+      alt: 'This is the cover of the book called Adventures of Huckleberry Finn',
+      description:
+        "The novel's preeminence derives from its wonderfully imaginative re-creation of boyhood adventures along the Mississippi River."
+    },
+    oliver_twist: {
+      author: 'Charles Dickens',
+      name: 'Oliver Twist',
+      alt: 'This is the cover of the book called Oliver Twist',
+      description:
+        'A profound social critic, Dickens introduced genteel readers to the problems of the poor in a way that had rarely been attempted before.'
+    },
+    frankenstein: {
+      author: 'Mary Shelley',
+      name: 'Frankenstein',
+      alt: 'This is the cover of the book called Frankenstein',
+      description:
+        "The story of Victor Frankenstein's terrible creation and the havoc it caused has enthralled generations of readers and inspired countless writers of horror and suspense. "
+    },
+    return_of_native: {
+      author: 'Thomas Hardy',
+      name: 'The Return of the Native',
+      alt: 'This is the cover of the book called The Return of the Native',
+      description:
+        'It first appeared in the magazine Belgravia, a publication known for its sensationalism, and was presented in twelve monthly installments from January to December 1878.'
+    }
+  };
+
+  const image = {
+    home_and_world: './images/home_and_world.jpeg',
+    the_rescue: './images/the_rescue.jpeg',
+    heart_of_darkness: './images/heart_of_darkness.jpeg',
+    country_of_pointed_fires: './images/country_of_pointed_fires.jpeg',
+    howards_end: './images/howards_end.jpeg',
+    robinson_crusoe: './images/robinson_crusoe.jpeg',
+    the_odyssey: './images/the_odyssey.jpeg',
+    death_in_venice: './images/death_in_venice.jpeg',
+    adventures_of_huckleberry_finn: './images/adventures_of_huckleberry_finn.jpeg',
+    oliver_twist: './images/oliver_twist.jpeg',
+    frankenstein: './images/frankenstein.jpeg',
+    return_of_native: './images/return_of_native.jpeg'
+  };
+
+  const div = document.getElementById('content');
+
+  const favoriteBooks = document.createElement('h1');
+  favoriteBooks.innerHTML = 'My Favorite Books';
+  div.appendChild(favoriteBooks);
+  favoriteBooks.className = 'favoriteBooks';
+
+  const divTag = document.createElement('div');
+  divTag.className = 'booksPart';
+
+  const ulTag = document.createElement('ul');
+  ulTag.className = 'books';
+
+  booksList.forEach(function(id) {
+    const img = document.createElement('img');
+    img.alt = information[id].alt;
+    img.src = image[id];
+    img.className = 'altName';
+
+    const liTag = document.createElement('li');
+    liTag.className = 'book';
+
+    const div2 = document.createElement('div');
+    div2.className = 'section';
+
+    const writer = document.createElement('h3');
+    writer.innerHTML = information[id].author;
+    writer.className = 'writer';
+
+    const title = document.createElement('h2');
```

Comment:
I see 9 places in total where you do:

```
const var = document.createElement('<tag>');
var.className = '.....';
```

That is in total 18 lines of code where we are repeating ourselves. There is a principle in software development named Don't Repeat Yourself (DRY). That means, if you see repeating yourself like in our case, we should take out the common pieces and move those to a separate function.

Thus, you can have a function say `setElement` which accepts some arguments and performs the same operation as many times as you want it to. Would you like to try it that way? You will find your code to be much cleaner and easier to read. 🙂

Reply:
I will try to make my home work look better :") 
thank you very much for review my homework .

Reply:
Hi
excuse me, I want to update my homework. But it give All checks have failed.
I don't know the problem exactly.
thanks